app/domain/errors.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+require 'util/error_class'
+
+unless defined? Errors::Authentication::AuthenticatorNotFound
+  # this wrapper is here so these classes will not be loaded by Rails
+  # auto-load. #TODO: fix this in a proper manner
+
+  module Errors
+    module Authentication
+
+      AuthenticatorNotFound = ::Util::ErrorClassWithCode.new(
+        msg: "'{0}' wasn't in the available authenticators",
```

Comment:
What does "**in** the authenticators" mean? just want to check the preposition is correct

Reply:
the available authenticators are all those which are implemented on conjur (oidc, ldap, k8s & iam)

Reply:
maybe change to "Authenticator '{0}' is not implemented in Conjur"?

Reply:
You could say "wasn't in the list of available authenticators" if you wanted to be explicit.  Oren your version works grammatically but it want to maintain the distinction between "available" (you've turned it on and configured it) vs "implemented" (the devs at Conjur have built it and it exists).  We could take your version but just change that word:

```
Authenticator '{0}' is not currently available
```

But honestly any of these wording sound ok to me.

Reply:
i disagree on the definition. 

the installed authenticators are defined:
```
def installed_authenticators
    @installed_authenticators ||= Authentication::InstalledAuthenticators.authenticators(ENV)
  end

def self.authenticators(env, authentication_module: ::Authentication)
      loaded_authenticators(authentication_module)
        .select { |cls| valid?(cls) }
        .map { |cls| [url_for(cls), authenticator_instance(cls, env)] }
        .to_h
    end
```

the way we mark an authenticator as valid is:
```
def valid?
        valid_name? && valid_parent_name? && valid_interface?
end

      def valid_name?
        own_name == 'Authenticator'
      end

      def valid_parent_name?
        parent_name =~ /^Authn/
      end

      def valid_interface?
        @cls.method_defined?(:valid?)
      end
```

So, in conclusion, the available/installed authenticators are those who are implemented in conjur code (they have an `Authn-somehting::Authenticator` class with a `valid?` method). so it is exactly what you call "implemented" (the devs at Conjur have built it and it exists) and not your definition of "available" (you've turned it on and configured it).

am i missing something?

Reply:
So then ho wabot "Authenticator '{0}' does not exist in Conjur"?

Reply:
@orenbm You are correct.  I read it wrong.

My reply was based on the assumption that this was the error raised when the authenticator wasn't in `CONJUR_AUTHENTICATORS`.  It looks like you're saying this is for when the authenticator doesn't even exist, so I like your message.